tx.go

```diff
@@ -84,7 +85,9 @@ func transaction(db *sql.DB, ctx context.Context, opts *sql.TxOptions, callback
 	}
 	if txErr != nil {
 		if err := sqlTx.Rollback(); err != nil {
-			return err
+			if !errors.Is(err, sql.ErrTxDone) {
```

Comment:
I dived into the sql's codes and found that `tx.Commit()` also returns `ErrTxDone` on the similar situation. I think error checking should be done more globally in L:86 block, or should we check context's cancelation before rollback/commit? For `ErrTxDone` seems to be returned on various situations (simply called twice), it's not desirable to omit it naively.

Reply:
Check `ctx.Done()` before rollback or commit.
In the case of `Rollback()`, we now handle `ErrTxDone` and return the original error, in case the`Rollback()` was unfortunately canceled before execution.
In the case of `Commit()`, the commit has failed, so we do not handle it and return the error as is.